bindings/src/main/java/io/pravega/storage/hdfs/HDFSChunkStorage.java

```diff
@@ -0,0 +1,473 @@
+     *
+     * @return True or false.
+     */
+    @Override
+    public boolean supportsAppend() {
+        return true;
+    }
+
+    /**
+     * Gets a value indicating whether this Storage implementation supports truncate operation on underlying storage object.
+     *
+     * @return True or false.
+     */
+    @Override
+    public boolean supportsTruncation() {
+        return false;
+    }
+
+    //endregion
+
+    //region AutoCloseable Implementation
+
+    @Override
+    public void close() {
+        if (!this.closed.getAndSet(true)) {
+            if (this.fileSystem != null) {
+                try {
+                    this.fileSystem.close();
+                    this.fileSystem = null;
+                } catch (IOException e) {
+                    log.warn("Could not close the HDFS filesystem: {}.", e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Retrieves the ChunkInfo for given name.
+     *
+     * @param chunkName String name of the storage object to read from.
+     * @return ChunkInfo Information about the given chunk.
+     * @throws ChunkStorageException Throws IOException in case of I/O related exceptions.
+     * @throws IllegalArgumentException If argument is invalid.
+     */
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        ensureInitializedAndNotClosed();
+        try {
+            FileStatus last = fileSystem.getFileStatus(getFilePath(chunkName));
+            ChunkInfo result = ChunkInfo.builder().name(chunkName).length(last.getLen()).build();
+            return result;
+        } catch (IOException e) {
+            throwException(chunkName, "doGetInfo", e);
+        }
+        return null;
+    }
+
+    /**
+     * Creates a new file.
+     *
+     * @param chunkName String name of the storage object to create.
+     * @return ChunkHandle A writable handle for the recently created chunk.
+     * @throws ChunkStorageException Throws IOException in case of I/O related exceptions.
+     * @throws IllegalArgumentException If argument is invalid.
+     */
+    @Override
+    protected ChunkHandle doCreate(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        ensureInitializedAndNotClosed();
+        try {
+            Path fullPath = getFilePath(chunkName);
+            // Create the file, and then immediately close the returned OutputStream, so that HDFS may properly create the file.
+            this.fileSystem.create(fullPath, READWRITE_PERMISSION, false, 0, this.config.getReplication(),
+                    this.config.getBlockSize(), null).close();
+            log.debug("Created '{}'.", fullPath);
+
+            // return handle
+            return ChunkHandle.writeHandle(chunkName);
+        } catch (org.apache.hadoop.fs.FileAlreadyExistsException e) {
```

Comment:
Can we import rather than have the full reference?

Reply:
fixed